cmd/internal/cli/actions.go

```diff
@@ -154,7 +154,16 @@ func handleShub(u string) (string, error) {
 	imageName := uri.GetName(u)
 	imagePath := cache.ShubImage("hash", imageName)
 
-	libexec.PullShubImage(imagePath, u, true, noHTTPS)
+	exists, err := cache.ShubImageExists("hash", imageName)
+	if err != nil {
+		return "", fmt.Errorf("unable to check if %v exists: %v", imagePath, err)
+	}
+	if !exists {
+		sylog.Infof("Downloading shub image")
+		libexec.PullShubImage(imagePath, u, true, noHTTPS)
+	} else {
+		sylog.Infof("Use image from cache")
```

Comment:
Same comment here, probably should be `Verbosef`. Does a user expect these to print when running without any logging level commands? What's your opinion here?

Reply:
For both of these changes, the behavior is consistent with the handleLibrary() and handleNet() routines.  I agree, at least for the "Use image from cache" log message that Verbosef() is better.  No need to have that spit out for normal use.

Should the other routines be changed as well for using the image in the cache? And for the "downloading image" message in all of the routines? Is it ok to change the existing default behavior for logging?

Reply:
>I agree, at least for the "Use image from cache" log message that Verbosef() is better. No need to have that spit out for normal use.

Why don't we go ahead and make that change then. I think it's not so useful to write that out

>Should the other routines be changed as well for using the image in the cache? And for the "downloading image" message in all of the routines? Is it ok to change the existing default behavior for logging?

I think it's okay if we change that. Maybe we will first merge this Pull Request as is, then you can follow up with a second pull request changing all of the logging messages? Are you up for that @michaelmoore10 ?

Reply:
Yep.  I'm up for that.